sensor.py

```diff
@@ -15,15 +16,28 @@
 async def async_setup_platform(hass, config, async_add_entities,
                                discovery_info=None):
     """Set up HomeSeer sensor-type devices."""
-    from pyhs3 import HASS_SENSORS, DEVICE_ZWAVE_BATTERY
+    from pyhs3 import (HASS_SENSORS, DEVICE_ZWAVE_BATTERY, DEVICE_ZWAVE_TEMPERATURE,
+                       DEVICE_ZWAVE_RELATIVE_HUMIDITY, DEVICE_ZWAVE_LUMINANCE,
+                       DEVICE_ZWAVE_FAN_STATE, DEVICE_ZWAVE_OPERATING_STATE)
 
     sensor_devices = []
     homeseer = hass.data[DOMAIN]
+    units = hass.config.units.name
 
     for device in homeseer.devices:
         if device.device_type_string in HASS_SENSORS:
             if device.device_type_string == DEVICE_ZWAVE_BATTERY:
                 dev = HSBattery(device, homeseer)
+            elif device.device_type_string == DEVICE_ZWAVE_TEMPERATURE:
+                dev = HSTemperature(device, homeseer, units)
+            elif device.device_type_string == DEVICE_ZWAVE_RELATIVE_HUMIDITY:
+                dev = HSHumidity(device, homeseer)
+            elif device.device_type_string == DEVICE_ZWAVE_LUMINANCE:
+                dev = HSLuminance(device, homeseer)
+            elif device.device_type_string == DEVICE_ZWAVE_FAN_STATE:
+                dev = HSFanState(device, homeseer)
+            elif device.device_type_string == DEVICE_ZWAVE_OPERATING_STATE:
+                dev = HSOperatingState(device, homeseer)
             else:
                 dev = HSSensor(device, homeseer)
```

Comment:
To clean this up we should create a factory method similar to the get_zwave_device in pyhs3 that takes the string as input and returns the appropriate object.

Reply:
I will make that change.